packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -498,6 +498,9 @@ export default class CodeGen {
 
         // TODO [#3370]: remove experimental template expression flag
         if (this.state.config.experimentalComplexExpressions) {
+            // Cloning here is necessary because `this.replace()` is destructive, and we might use the
+            // node later during static content optimization
+            expression = doStructuredClone(expression);
```

Comment:
@ekashida Do you know if we still need the `expression = doStructuredClone(expression);` below, if we're doing it here already?

Reply:
We could hoist it out of the `if` and have just one call to `doStructuredClone()` but we'll be refactoring this whole routine soon so I'll just leave it as is for now.

Reply:
👍 sounds good!